### DISCUSSION TITLE
Pattern question on disabling object transfer

### DISCUSSION THREAD

**Discussion by 0x-j** [aptos-labs/aptos-developer-discussions#175] [Questions] [answered]
### Describe your question in detail.

When creating an object that stores some custom data, do we suggest developers to always disable object transfer when transfer is not expected? 

For example, I have a `User` struct that stores `username`. I create a `User` resource in a new object for each user of my app. I don't think anyone would transfer the user object, also the transfer won't affect my app becaue my app only looks at `username` field instead of object owner. THis is the case disable transfer is unnecessary. But 

### What have you tried or looked at?  Or how can we reproduce the error?

N/A

### Which operating system are you using?

N/A

### Which SDK or tool are you using? (if any)

N/A

**Comment by kshitijc1506** [comment, marked as answer]
I'm curious, what's the benefit of allowing transfers? I think the main reason for disabling transfer would be unintended consequences of an accidental transfer. 

For example, the new (say malicious) object owner can now modify the username, which the original owner (the actual user) may or may not want. In decentralized social applications, custody of one's username is a theme that makes the concept appealing. No one else should be able to manipulate the data that you own, and disabling transfers would fortify that idea. 

However, in a more gamified approach this _could_ make sense. For instance if there was some sort of value tied to the User struct, and people would like to be able to trade this "pseudo-identity". This kind of flexibility also sounds pretty "crypto native" to me. Then again, this sounds like a feature made for the DA standard. 

I think to prevent unintended consequences you can also just add module logic (i.e. centralized list of usernames). Although this seems like a more EVM-type approach to me.

**Reply by 0x-j** [reply]
Good point, for simplicity I'll disable transfer unless it's needed.

**Reply by gregnazario** [reply]
It's really dependent on the use case.  Is the object one of a kind that provides access to something else?

Maybe don't want to transfer it.

Is it for an NFT?  Then you probably want transfer on, unless it's soulbound.

This just provides general flexibility
